packages/Notification/__tests__/Notification.spec.jsx

```diff
@@ -135,4 +135,27 @@ describe('<Notification />', () => {
       expect(icon.props().variant).toBeUndefined()
     })
   })
+  describe('onDismiss', () => {
+    it('calls the provided callback when dismissible', () => {
+      const onDismiss = jest.fn()
+      const notification = doMount({
+        variant: 'error',
+        dismissible: true,
+        dismissibleA11yLabel: 'Close',
+        onDismiss,
+      })
+      notification.find(StandaloneIcon).simulate('click')
+      expect(onDismiss).toHaveBeenCalled()
+    })
+    it('does not call the provided callback when the dismissible prop is false', () => {
```

Comment:
This test does not do what it says.

Reply:
I think this test can be replaced with a detection that the `warn` has been fired when `{{ dismissible: false, onDismiss }}` is configured. 

There is already a test above that checks to make sure `StandaloneIcon` doesn't exist when `dismissible: false`.